spec/index.bs

```diff
@@ -3156,23 +3154,29 @@ A {{GPUProgrammableStageDescriptor}} describes the entry point in the user-provi
-            the |binding| has to be a non-comparison sampler.
-        1. If |entry|.{{GPUBindGroupLayoutEntry/type}} is {{GPUBindingType/"comparison-sampler"}},
-            the |binding| has to be a comparison sampler.
-        1. If |entry|.{{GPUBindGroupLayoutEntry/type}} is {{GPUBindingType/"sampled-texture"}},
-            the |binding| has to be a sampled texture with the component type of |entry|.{{GPUBindGroupLayoutEntry/textureComponentType}},
-            and it must be multisampled if and only if |entry|.{{GPUBindGroupLayoutEntry/multisampled}} is true.
-        1. If |entry|.{{GPUBindGroupLayoutEntry/type}} is {{GPUBindingType/"readonly-storage-texture"}},
-            the |binding| has to be a read-only storage texture with format of |entry|.{{GPUBindGroupLayoutEntry/storageTextureFormat}}.
-        1. If |entry|.{{GPUBindGroupLayoutEntry/type}} is {{GPUBindingType/"writeonly-storage-texture"}},
-            the |binding| has to be a writable storage texture with format of |entry|.{{GPUBindGroupLayoutEntry/storageTextureFormat}}.
-        1. If |entry|.{{GPUBindGroupLayoutEntry/type}} is {{GPUBindingType/"uniform-buffer"}},
-            the |binding| has to be a uniform buffer.
-        1. If |entry|.{{GPUBindGroupLayoutEntry/type}} is {{GPUBindingType/"storage-buffer"}},
-            the |binding| has to be a storage buffer.
-        1. If |entry|.{{GPUBindGroupLayoutEntry/type}} is {{GPUBindingType/"readonly-storage-buffer"}},
-            the |binding| has to be a read-only storage buffer.
+        1. If |entry|.{{GPUBindGroupLayoutEntry/type}} is:
+            <dl class=switch>
+
+              : {{GPUBindingType/"sampler"}}
+              :: the |binding| has to be a non-comparison sampler
+              : {{GPUBindingType/"comparison-sampler"}}
+              :: the |binding| has to be a comparison sampler
+              : {{GPUBindingType/"sampled-texture"}}
+              ::
+                1. the |binding| has to be a sampled texture with the component type of |entry|.{{GPUBindGroupLayoutEntry/textureComponentType}}.
+                2. it must be multisampled if and only if |entry|.{{GPUBindGroupLayoutEntry/multisampled}} is true.
+              : {{GPUBindingType/"readonly-storage-texture"}}
+              :: the |binding| has to be a read-only storage texture with format of |entry|.{{GPUBindGroupLayoutEntry/storageTextureFormat}}
+              : {{GPUBindingType/"writeonly-storage-texture"}}
+              :: the |binding| has to be a writable storage texture with format of |entry|.{{GPUBindGroupLayoutEntry/storageTextureFormat}}
+              : {{GPUBindingType/"uniform-buffer"}}
+              :: the |binding| has to be a uniform buffer
+              : {{GPUBindingType/"storage-buffer"}}
+              :: the |binding| has to be a storage buffer
+              : {{GPUBindingType/"readonly-storage-buffer"}}
+              :: the |binding| has to be a read-only storage buffer
+
```

Comment:
nit: 4-space indent should avoid the need for this and other blank lines in this block